secure-api-gateway-ob-uk-rcs-server/src/main/java/com/forgerock/sapi/gateway/ob/uk/rcs/server/api/ConsentDetailsApiController.java

```diff
@@ -149,31 +157,6 @@ public ResponseEntity<ConsentDetails> getConsentDetails(String consentRequestJws
         }
     }
 
-    /*
-        DebtorAccount is optional, but the PISP could provide the account identification details for the PSU.
-        If the account identifier match with an existing one for that psu then we update the debtor account with the proper accountId
-        and set the debtor account with balance as accounts to be display in the consent UI
-     */
-    private void setDebtorAccountWithBalance(PaymentsConsentDetails details, String consentRequestJws, String intentId) {
-        FRAccountIdentifier debtorAccount = details.getDebtorAccount();
-        if (Objects.nonNull(debtorAccount)) {
-            FRAccountWithBalance accountWithBalance = accountService.getAccountWithBalanceByIdentifiers(
-                    details.getUserId(), debtorAccount.getName(), debtorAccount.getIdentification(), debtorAccount.getSchemeName()
-            );
-            if (Objects.nonNull(accountWithBalance)) {
-                debtorAccount.setAccountId(accountWithBalance.getAccount().getAccountId());
-                details.setAccounts(List.of(accountWithBalance));
-            } else {
-                String message = String.format("Invalid debtor account provide in the consent for the intent ID: '%s', the debtor account provided in the consent doesn't exist", intentId);
-                log.error(message);
-                throw new InvalidConsentException(consentRequestJws, ErrorType.ACCOUNT_SELECTION_REQUIRED,
-                        OBRIErrorType.REQUEST_BINDING_FAILED, message,
-                        details.getClientId(),
-                        intentId);
-            }
-        }
-    }
```

Comment:
Moved to DebtorAccountService class so that it can be reused by old and new details impls